### ISSUE TITLE
Change the names of islinear and ismultilinear

### ISSUE THREAD

**Issue by siddharthist** [UniMath/UniMath#839]
Part of this is my own fault. Generally, one thinks of a "linear" function as one that preserves scalar multiplication and addition. Currently, our definition speaks only to scalar multiplication. 

Additionally, as adding "multi" as a prefix to every name seemed like a good way to name things in `Multimodules.v`, we have a definition of "multilinear" functions that are not generally what one means when one speaks of mulitilinear functions in linear algebra. 

I'm not sure what the best names would be, but perhaps something like `preserves_scalar_mult` or `preserves_rng_action`?

**Comment by DanGrayson** [comment]
Bourbaki calls them multimodule homomorphisms, so what about just `isHomomorphism`?

**Comment by siddharthist** [comment]
There are two complications with that idea: 
 1. Right now, there is a separate definition of `is(multi)linear` and `(multi)modulefun`, with a `(multi)modulefun` being a function that `is(multi)linear` and a `monoidfun`. Thus, a `(multi)modulefun` is a morphism of modules, _not_ a `linearfun`. 
2. What if a user is working with modules and some other algebraic structure, and both define `isHomomorphism`? I think the convention of UniMath to include the name of the structure in question is a good practice, and we should try to follow it here.

**Comment by DanGrayson** [comment]
(1) Yes, but your definition of `islinearfun` is wrong if it doesn't mention compatibility with addition.

(2) I see no reason to include type names in function names.  In case of ambiguity, the standard was to resolve it is by qualifying the name further, e.g., `Multimodule.isHomomorphism`.   Any other way just seems to be clutter.  @benediktahrens may have a different opinion.

By the way, Bourbaki's definition of multimodule is not the same as yours -- they let one list of rings act on the left and another list of rings act on the right.  Why not follow the textbook?

**Comment by benediktahrens** [comment]
On 12/18/2017 12:15 AM, Langston Barrett wrote:
> There are two complications with that idea: 1. Right now, there is a
> separate definition of `is(multi)linear` and `(multi)modulefun`, with
> a `(multi)modulefun` being a function that `is(multi)linear` and a
> `monoidfun`. Thus, a `(multi)modulefun` is a morphism of modules,
> _not_ a `linearfun`. 2. What if a user is working with modules and
> some other algebraic structure, and both define `isHomomorphism`? I
> think the convention of UniMath to include the name of the structure
> in question is a good practice, and we should try to follow it here.

There are many coercions in the algebra library, so that it can be 
difficult to figure out as what a given object is considered in a given 
type.
E.g., a ring might be considered as anything less than a ring at a given 
moment, if Coq sees the need for that to make something well-typed.

This seems to me an argument to include the name of the structure in the 
name for morphisms.

That being said, I don't use the Algebra package, so I am not sure if 
that can happen in practice.

**Comment by siddharthist** [comment]
@DanGrayson On the subject of having two definitions, I followed the conventions set up in the precious definition of a module. I agree with you that it makes no sense, so let's combine them as you suggest.

I didn't use two lists because one is extraneous - if you want something to be in the "left" list, include as usual, and if you want a ring to be in the "right" list, simply include its opposite.

**Comment by siddharthist** [comment]
Ah, I didn't mean to close this

**Comment by DanGrayson** [comment]
Yes, one of the lists is extraneous, and Bourbaki points that out.  Nevertheless, Bourbaki chose to have both of them.  The reason must be that the most useful example of a multimodule is that of an R-S-bimodule, where R acts on the left and S acts on the right.  Yes, it's the same as an (R,S^op)-()-multimodule, but that circumlocution is worth avoiding, because it will complicate the main applications of this notion.
